jpyinterpreter/src/main/java/org/optaplanner/jpyinterpreter/PythonBytecodeToJavaBytecodeTranslator.java

```diff
@@ -1102,6 +1098,96 @@ private static void translatePythonBytecodeToMethod(MethodDescriptor method, Str
         methodVisitor.visitEnd();
     }
 
+    public static void writeInstructionsForOpcodes(FunctionMetadata functionMetadata,
+            List<StackMetadata> stackMetadataForOpcodeIndex, List<Opcode> opcodeList) {
+        writeInstructionsForOpcodes(functionMetadata, stackMetadataForOpcodeIndex, opcodeList, ignored -> {
+        });
+    }
+
+    public static void writeInstructionsForOpcodes(FunctionMetadata functionMetadata,
+            List<StackMetadata> stackMetadataForOpcodeIndex, List<Opcode> opcodeList,
+            Consumer<PythonBytecodeInstruction> runAfterLabelAndBeforeArgumentors) {
+        PythonCompiledFunction pythonCompiledFunction = functionMetadata.pythonCompiledFunction;
+        MethodVisitor methodVisitor = functionMetadata.methodVisitor;
+        Map<Integer, Label> bytecodeCounterToLabelMap = functionMetadata.bytecodeCounterToLabelMap;
+        Map<Integer, List<Runnable>> bytecodeIndexToArgumentorsMap = functionMetadata.bytecodeCounterToCodeArgumenterList;
+
+        Map<Integer, List<Runnable>> exceptionTableTryBlockMap = new HashMap<>();
+        Map<Integer, Label> exceptionTableStartLabelMap = new HashMap<>();
+        Map<Integer, Label> exceptionTableTargetLabelMap = new HashMap<>();
+        Map<Integer, AtomicReference<int[]>> tryBlockStartToStoredStackMap = new HashMap<>();
+
+        for (PythonExceptionTable.ExceptionBlock exceptionBlock : pythonCompiledFunction.co_exceptiontable.getEntries()) {
+            if (exceptionBlock.blockStartInstructionInclusive == exceptionBlock.blockEndInstructionExclusive) {
+                continue; // Empty try block range
+            }
+            AtomicReference<int[]> storedStack = tryBlockStartToStoredStackMap
+                    .computeIfAbsent(exceptionBlock.blockStartInstructionInclusive, key -> new AtomicReference<>());
```

Comment:
Any particular reason for `AtomicReference` here?
`int[]` already gives you an effectively-final reference, so there must be some other reason I'm missing.